extensions/pkg/controller/utils.go

```diff
@@ -150,3 +151,17 @@ func ShouldSkipOperation(operationType gardencorev1beta1.LastOperationType, obj
 func GetObjectByReference(ctx context.Context, c client.Client, ref *autoscalingv1.CrossVersionObjectReference, namespace string, obj client.Object) error {
 	return c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: v1beta1constants.ReferencedResourcesPrefix + ref.Name}, obj)
 }
+
+// IsShootNamespace returns true if the namespace has the garden role shoot label.
+func IsShootNamespace(ctx context.Context, reader client.Reader, namespace string) (bool, error) {
```

Comment:
There is already a `isShootNamespace` function available (not yet public) - see [here](https://github.com/gardener/gardener/blob/bcae3ebbbb21a88732a7937c3fcd4cd0672f85a2/pkg/component/kubernetes/apiserverexposure/service.go#L223). WDYT about reusing it?